[E2E] Operator Pod is never ready

Last e2e sessions have all failed with the following error:

```
01:34:53 {
	"log.level": "error",
	"@timestamp": "2021-02-15T00:34:52.722Z",
	"log.logger": "e2e-341ep",
	"message": "Failed to run e2e tests",
	"service.version": "0.0.0-SNAPSHOT+00000000",
	"service.type": "eck",
	"ecs.version": "1.4.0",
	"error": "operator pod `e2e-341ep-operator-0` not ready",
	"error.stack_trace": "github.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897\nmain.main\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/cmd/main.go:31\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:204"
}
01:34:53  Error: operator pod `e2e-341ep-operator-0` not ready
01:34:53  exit status 1
01:34:53  make: *** [Makefile:431: e2e-run] Error 1
```

I manually ran a test and it looks like the image name of the operator is wrong:

```
      state:
        waiting:
          message: Back-off pulling image "docker.elastic.co/eck-snapshots/eck-operator-ci:1.5.0-SNAPSHOT-c20894e0"
          reason: ImagePullBackOff
```

While I think it should be `docker.elastic.co/eck-snapshots/eck-operator:1.5.0-SNAPSHOT-2021-02-15-c20894e`

--- a/Makefile
+++ b/Makefile
@@ -429,8 +429,7 @@ e2e-docker-multiarch-build: clean
 
 e2e-run:
 	@go run test/e2e/cmd/main.go run \
-		--operator-image-repo=$(BASE_IMG) \
-		--operator-image-tag=$(IMG_VERSION) \
+		--operator-image=$(OPERATOR_IMAGE) \
 		--e2e-image=$(E2E_IMG) \
 		--test-regex=$(TESTS_MATCH) \
 		--test-license=$(TEST_LICENSE) \

--- a/test/e2e/cmd/run/command.go
+++ b/test/e2e/cmd/run/command.go
@@ -25,8 +25,7 @@ type runFlags struct {
 	e2eImage              string
 	elasticStackVersion   string
 	kubeConfig            string
-	operatorImageRepo     string
-	operatorImageTag      string
+	operatorImage         string
 	testLicensePKeyPath   string
 	testContextOutPath    string
 	testLicense           string
@@ -85,8 +84,7 @@ func Command() *cobra.Command {
 	cmd.Flags().StringVar(&flags.kubeConfig, "kubeconfig", "", "Path to kubeconfig")
 	cmd.Flags().BoolVar(&flags.local, "local", false, "Create the environment for running tests locally")
 	cmd.Flags().StringSliceVar(&flags.managedNamespaces, "managed-namespaces", []string{"mercury", "venus"}, "List of managed namespaces")
-	cmd.Flags().StringVar(&flags.operatorImageRepo, "operator-image-repo", "", "Operator image repo")
-	cmd.Flags().StringVar(&flags.operatorImageTag, "operator-image-tag", "", "Operator image tag")
+	cmd.Flags().StringVar(&flags.operatorImage, "operator-image", "", "Operator image")
 	cmd.Flags().IntVar(&flags.operatorReplicas, "operator-replicas", 1, "Operator replicas")
 	cmd.Flags().BoolVar(&flags.skipCleanup, "skip-cleanup", false, "Do not run cleanup actions after test run")
 	cmd.Flags().StringVar(&flags.testContextOutPath, "test-context-out", "", "Write the test context to the given path")

--- a/test/e2e/cmd/run/run.go
+++ b/test/e2e/cmd/run/run.go
@@ -131,6 +131,11 @@ func (h *helper) createScratchDir() error {
 }
 
 func (h *helper) initTestContext() error {
+	imageParts := strings.Split(h.operatorImage, ":")
+	if len(imageParts) != 2 {
+		return fmt.Errorf("invalid operator image: %s", h.operatorImage)
+	}
+
 	h.testContext = test.Context{
 		AutoPortForwarding:  h.autoPortForwarding,
 		E2EImage:            h.e2eImage,
@@ -147,8 +152,9 @@ func (h *helper) initTestContext() error {
 			ManagedNamespaces: make([]string, len(h.managedNamespaces)),
 			Replicas:          h.operatorReplicas,
 		},
-		OperatorImageRepo:     h.operatorImageRepo,
-		OperatorImageTag:      h.operatorImageTag,
+		OperatorImage:         h.operatorImage,
+		OperatorImageRepo:     imageParts[0],
+		OperatorImageTag:      imageParts[1],
 		TestLicense:           h.testLicense,
 		TestLicensePKeyPath:   h.testLicensePKeyPath,
 		MonitoringSecrets:     h.monitoringSecrets,

--- a/test/e2e/test/context.go
+++ b/test/e2e/test/context.go
@@ -90,6 +90,7 @@ type Context struct {
 	E2EServiceAccount     string            `json:"e2e_service_account"`
 	ElasticStackVersion   string            `json:"elastic_stack_version"`
 	LogVerbosity          int               `json:"log_verbosity"`
+	OperatorImage         string            `json:"operator_image"`
 	OperatorImageRepo     string            `json:"operator_image_repo"`
 	OperatorImageTag      string            `json:"operator_image_tag"`
 	TestLicense           string            `json:"test_license"`

